app/src/main/java/com/example/androidDeviceDetails/cooker/LocationCooker.kt

```diff
@@ -21,52 +23,52 @@ class LocationCooker : BaseCooker() {
         GlobalScope.launch {
             val res = RoomDB.getDatabase()!!.locationDao()
                 .readDataFromDate(time.startTime, time.endTime) as ArrayList<LocationModel>
-            if (res.isNotEmpty()) {
-                val processedData = processData(res)
-                val cookedData = cookProcessedData(processedData)
-                iCookingDone.onComplete(cookedData as ArrayList<T>)
-            } else {
+            if (res.isNotEmpty())
+                iCookingDone.onComplete(cookData(res) as ArrayList<T>)
+            else
                 iCookingDone.onComplete(arrayListOf())
-            }
         }
     }
 
-    private fun processData(locationList: ArrayList<LocationModel>): HashMap<String, LocationData> {
-        val processedLocations = HashMap<String, LocationData>()
+    private fun cookData(locationList: ArrayList<LocationModel>): ArrayList<LocationData> {
+        val geoHashMap = HashMap<String, LocationData>()
+        val countMap = HashMap<String, Int>()
         var preLoc = locationList.first()
-        val firstHash =
-            GeoHash.encodeHash(preLoc.latitude, preLoc.longitude, LocationConstants.GEO_HASH_LENGTH)
-        processedLocations[firstHash] = LocationData(preLoc.latitude, preLoc.longitude, 1, "", 0)
+        var prevHash = getGeoHash(preLoc.latitude, preLoc.longitude)
+        geoHashMap[prevHash] = LocationData(preLoc.latitude, preLoc.longitude, 1, "", 0)
+        countMap[prevHash] = 1
         for (loc in locationList.subList(1, locationList.size)) {
-            val geoHash =
-                GeoHash.encodeHash(loc.latitude, loc.longitude, LocationConstants.GEO_HASH_LENGTH)
-            if (loc.latitude != preLoc.latitude && loc.longitude != preLoc.longitude) {
-                if (geoHash in processedLocations.keys) {
-                    processedLocations[geoHash]!!.latitude += loc.latitude
-                    processedLocations[geoHash]!!.longitude += loc.longitude
-                    processedLocations[geoHash]!!.count += 1
-                    processedLocations[geoHash]!!.totalTime += loc.timeStamp - preLoc.timeStamp
-                } else {
-                    processedLocations[geoHash] = LocationData(
-                        loc.latitude, loc.longitude, 1, "", loc.timeStamp - preLoc.timeStamp
-                    )
+            val newHash = getGeoHash(loc.latitude, loc.longitude)
+            if (newHash in geoHashMap.keys) {
+                countMap[newHash] = countMap[newHash]!! + 1
+                geoHashMap[newHash]?.run {
+                    latitude += loc.latitude
```

Comment:
Can we do this like :
latitude = (latitude + loc.latitude)/2
longitude = (longitude + loc.longitude )/2

So we can remove the cookProcessedData( ) completely?